charts/govuk-rails-app/templates/assets-upload-job.yaml

```diff
@@ -0,0 +1,22 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ .Release.Name }}-uploadRailsAssets
+  annotations:
+    argocd.argoproj.io/hook: PreSync
+    argocd.argoproj.io/hook-delete-policy: HookSucceeded
+spec:
+  template:
+    spec:
+      containers:
+      - name: {{ .Release.Name }}-uploadRailsAssets
+        image: "{{ required "A valid .Values.appImage.repository entry required!" .Values.appImage.repository }}:{{ .Values.appImage.tag }}"
+        command:
+        - "bash"
+        - "-c"
+        - |
+          apt-get update && \
+          apt-get install -y awscli && \
```

Comment:
Would there be a way to use a pre-built image (perhaps the [official awscli image](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2-docker.html)) rather than using `apt-get install` during the deploy?

Reply:
This is the tricky part of doing the upload on deploy - it needs to get the assets from the built app image. Open to better suggestions

Reply:
Yes tricky one! I'm not entirely against adding the AWS CLI to the images, but if we need to do it, we should do it in the Dockerfile.

Did you already consider doing the upload at build time instead? We push artefacts to ECR from GitHub Actions, we could add a build step to a GitHub Action to push the assets up. I don't think this necessarily means we'd need to install `awscli` into the image, there might be a way to add a second step to the GitHub Action which reads from the output of the first step 🤔

Alternatively, we could do something with an init container in Kubernetes 🤔 

We built this previously into our deploy pipeline using Concourse:
https://github.com/alphagov/govuk-infrastructure/blob/ecs/concourse/pipelines/deploy.yml#L829-L837

The way this worked was we'd pass the assets directory as an output from the app image, and then use a `cli` image to push them up to S3.

Reply:
Sounds like a good plan. Thanks @sengi, I'm happy for us to merge it as-is to get it working 😃